docs/articles/blog/2020-7-15-introducing-the-testcafe-jenkins-plugin.md

```diff
@@ -0,0 +1,170 @@
+    "browsers": [
+        "chromium:headless:emulation:device=iPhone 4",
+        "chromium:headless:emulation:device=iPad Mini",
+        "chromium:headless"
+    ],
+    "src": "test/e2e/**/*",
+    "screenshots": {
+        "takeOnFails": true,
+        "fullPage": true
+    },
+    "videoPath": "videos",
+    "reporter": {
+        "name": "jenkins",
+        "output": "report.xml"
+    },
+    "appCommand": "node server.js"
+}
+```
+
+The *Browser* array contains the list of browsers TestCafe will use to run our test. The [TestCafe browser syntax](https://devexpress.github.io/testcafe/documentation/guides/concepts/browsers.html#use-chromium-device-emulation) allows us to specify the mobile devices we want Chromium to emulate. 
+
+The *name* property of the *reporter* object is set as *jenkins*, which ensures that the reports generated by TestCafe can be properly parsed by the server.
+
+If you decide against using a config file, don't forget to [manually set](https://devexpress.github.io/testcafe/documentation/reference/command-line-interface.html#-r-nameoutput---reporter-nameoutput) the report format when launching your tests, like so:
+
+```bash
+testcafe chrome test.js -r jenkins:report.xml
+```
+
+Otherwise, just run the following command:
+
+```bash
+npx testcafe
+```
+
+### Step 5. Configure the reporter
+
+Turn on the [JUnit](https://plugins.jenkins.io/junit/) reporter plugin: add the "Publish JUnit test result report" *post-build action*. To display screenshots and videos alongside your test results, select the option "Include links to TestCafe artifacts" from the "Additional test report features" drop-down.
+
+Important: Don't forget to check the "Retain long standard output/error" box. Otherwise the plugin will automatically truncate information concerning test screenshots and videos.
+
+![JUnit reporter configuration screen](/testcafe/images/jenkins-plugin/junit-reporter-configuration.png)
+
+### Step 6. Run the build
+
+Save the changes to your project and press the "Build now" button on the project page:
+
+![Jenkins Project menu with the Build Now Button highlighted](/testcafe/images/jenkins-plugin/build-now-button.png)
+
+## Review the results
+
+After the tests are completed, press the following link on the build page to view the results:
+
+![Test results link](/testcafe/images/jenkins-plugin/test-results-link.png)
+
+There you'll see links to screenshots and videos taken during the test.
+
+![Test results view](/testcafe/images/jenkins-plugin/test-results-view.png)
+
+These artifacts instantly indicate if the Cart button is present on the page.
+
+## Jenkins Pipeline integration
+
+If you don't want to use the Jenkins GUI to launch these tests, you can always make use of [Jenkins Pipeline](https://www.jenkins.io/doc/book/pipeline/). It lets you declaratively configure your CI/CD setup via a Jenkinsfile, a text file checked into the project's source control repository. Here's a sample Jenkinsfile to be used with TestCafe:
```

Comment:
Can you clarify what you mean by "checked into"? Is "checked" the correct technical term in this context?

Reply:
'Checking a file into source control' means instructing a source control management system of your choice to track this file going forward.

https://git-scm.com/book/en/v2/Customizing-Git-Git-Attributes : `<...> tell Git how to diff non-text files, or have Git filter content before you check it into or out of Git.`

https://devcenter.heroku.com/articles/git: `Common causes of large repositories are binary files checked into the repository  (Git is notoriously bad at handling binaries) or constantly-changing development logs.`

Reply:
Thanks for the clarification. Please see my new suggestion below.

Reply:
```suggestion
If you do not wish to use the Jenkins GUI to launch these tests, you can always use the [Jenkins Pipeline](https://www.jenkins.io/doc/book/pipeline/). It lets you declaratively configure your CI/CD setup via a Jenkinsfile - a text file checked into the project's source control repository. The following is a sample Jenkinsfile to be used with TestCafe:
```